cg/services/illumina/file_parsing/models.py

```diff
@@ -28,3 +29,41 @@ class DemuxMetrics(BaseModel):
     lane: int = Field(..., alias=DemuxMetricsColumnNames.LANE)
     sample_internal_id: str = Field(..., alias=DemuxMetricsColumnNames.SAMPLE_INTERNAL_ID)
     read_pair_count: int = Field(..., alias=DemuxMetricsColumnNames.READ_PAIR_COUNT)
+
+
+class DsmcEncryptionKey(BaseModel):
+    """Model representing the response from a PDC query."""
+
+    date: str
```

Comment:
So what we want to use this field as is a `datetime` object, not a string. Hence this type hint is wrong, given that whenever the `parse_date` validator has kicked in, this will not be a string. Using the "Annotated" pattern, we could write this as (when we have moved the `parse_date` method to be its standalone method)

```suggestion
    date: Annotated[datetime, BeforeValidator(parse_date)]
```

This has two major benefits - the type hint is correct, i.e. we say that it is a datetime object; and we can immediately see that the validation performed on the field is that it is parsed as a datetime object.